src/python/pants/init/engine_initializer.py

```diff
@@ -278,7 +270,7 @@ def build_configuration_singleton() -> BuildConfiguration:
 
         @rule
         def symbol_table_singleton() -> SymbolTable:
```

Comment:
FYI I tried deleting this singleton, but it resulted in every single rule failing. I'm confused why we need this - no `@rule` ever uses it.

Reply:
... that is indeed mysterious. I still need to clean up RuleGraph errors, but that will definitely be post 2.0, pre 2.1 polish i think.

Reply:
I couldn't figure this out either.